File: 1929-concatenation-of-array/1929-concatenation-of-array.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> out(2 * nums.size());
        int i = 0;
        for(i = 0; i < nums.size(); i++)
        {
            out[i] = nums[i];
            out[nums.size() + i] = nums[i];
        }
        return out;
    }
};

File: 1929-concatenation-of-array/1929-concatenation-of-array.go
func getConcatenation(nums []int) []int {
    if len(nums) == 0 {
        return []int{}
    } 
    ans := make([]int, len(nums)*2)
    for i := 0; i < len(ans); i++ {
        ans[i] = nums[i % len(nums)]
    }
    return ans
}